spyder/plugins/ipythonconsole.py

```diff
@@ -477,15 +477,42 @@ def setup_page(self):
         run_file_group.setLayout(run_file_layout)
         
         # ---- Advanced settings ----
+        # Enable Jedi completion in Ipyhton Console
+        ipy_jedi_group = QGroupBox(_("Jedi completion in Ipython console"))
+        ipy_jedi_label = QLabel(_("Enable Jedi based <tt>Tab</tt> completion in"
+                                "the Ipython console. E.g. completion of"
+                                "nested lits etc. This is similar to the "
+                                "greedy compiler, but without evaluating the code<br>"
+                                "<b>Attention:</b> This can slow down your console "
+                                "interaction when working with large dataframes!"))
+        ipy_jedi_label.setWordWrap(True)
+        ipy_jedi_box = newcb(_("Use Jedi Completer in Ipython"), "ipy_jedi_completer",
+                           tip="<b>Warning</b>: This can slow down your console"
+                                "when working with large dataframes ")
+        
+        ipy_jedi_layout = QVBoxLayout()
+        ipy_jedi_layout.addWidget(ipy_jedi_label)
+        ipy_jedi_layout.addWidget(ipy_jedi_box)
+        ipy_jedi_group.setLayout(ipy_jedi_layout)
+                
         # Greedy completer group
-        greedy_group = QGroupBox(_("Greedy completion"))
+        greedy_group = QGroupBox(_("Greedy completion in Ipython console"))
         greedy_label = QLabel(_("Enable <tt>Tab</tt> completion on elements "
                                 "of lists, results of function calls, etc, "
                                 "<i>without</i> assigning them to a "
                                 "variable.<br>"

```

Comment:
Should probably eliminate this ``<br>``, it looks awkward and takes up an extra line especially with the additional warning statement.